### DISCUSSION TITLE
Reading energy consumption and generation data

### DISCUSSION THREAD

**Discussion by TRIROG** [raomin/ESPAltherma#289] [Q&A]
Hi!
Altherma third generation (eg. Altherma 3 H MT) support energy consumption and energy generation data tracking for heating, cooling(if supported),  DHW and sum on a daily, weekly, monthly yearly basis and total since instalation . 

Since this data is available from user interface on internal unit i wonder does anyone know which registers this data is stored in?

**Comment by TRIROG** [comment]
*still interested in this*

**Comment by Arnold-n** [comment]
In my understanding the monthly/yearly information is calculated and stored in the wall room controller, not in the Daikin indoor unit. The Daikin only reports totals via the P1/P2 interface, and only upon explicit request by the wall room controller; the wall room controller makes and remembers the calculations for monthly/yearly.

**Reply by TRIROG** [reply]
Not sure i understand. Controller is integrated in my internal unit. But ok logically speaking it's likely a separate unit.  
But still  if this data is reported to controller, is there any way for espaltherma to also read and report this data?

**Reply by Arnold-n** [reply]
Indeed it's a separate controller. It is connected to the main system via P1/P2. I haven't found a way to get a controller to output such data (other than the overall counters). Whether the overall counters are accessible via X10A/espAltherma is an open question.
